### PR TITLE
Add spec for PopulationCountOp

### LINKED ISSUE #435: Add spec for PopulationCountOp

### PR DESCRIPTION
**Pull request by ghpvnist**
closes #435

### CODE REVIEW

File: docs/spec_draft.md [hunk 1]
```diff
@@ -1410,6 +1411,39 @@ More formally, `result[i0, ..., iR-1]` is equal to:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.popcnt
+
+### Semantics
+
+Counts the number of bits set in `operand` tensor element-wise and produces a
+`result` tensor.
+
```
Comment (subhankarshah) [inline docs/spec_draft.md:1420]: Performs element-wise count of the number of bits set in the `operand` and produces a `result` tensor.

or

Counts the number of bits set in `operand` tensor, element-wise, and produces a
`result` tensor.

File: docs/spec_draft.md [hunk 2]
```diff
@@ -1410,6 +1411,39 @@ More formally, `result[i0, ..., iR-1]` is equal to:
...
+### Constraints
+
+  * (C1) `operand` and `result` have the same type.
+
+### Examples
+
+```mlir
+// %operand: [0, 1, 2, 127]
+%result = "stablehlo.popcnt"(%lhs, %rhs) : (tensor<4xi8>) -> tensor<4xi8>
```
Comment (subhankarshah) [inline docs/spec_draft.md:1441]: %lhs %rhs -> %operand

Comment (subhankarshah) [reply docs/spec_draft.md:1441]: %result = "stablehlo.popcnt"(% operand) : (tensor<4xi8>) -> tensor<4xi8>

**Review by subhankarshah — CHANGES_REQUESTED** [review]
Please resolve minor comments

File: docs/spec_draft.md (hunk 2 above)
Comment (ghpvnist) [reply docs/spec_draft.md]: Thanks for the catch!

File: stablehlo/dialect/StablehloOps.td [hunk 3]
```diff
@@ -484,15 +484,15 @@ def StableHLO_PopulationCountOp: StableHLO_UnaryElementwiseOp<"popcnt",
...
-    See
-    https://www.tensorflow.org/xla/operation_semantics#element-wise_unary_functions.
+    See:
+    https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md#stablehlopopcnt
 
     Example:
-
     ```mlir
-    %0 = stablehlo.popcnt %arg0 : tensor<2x2xi32>
+    %result = "stablehlo.popcnt"(%operand) : (tensor<4xi8>) -> tensor<4xi8>
```
Comment (subhankarshah) [inline stablehlo/dialect/StablehloOps.td:495]: if pretty print is there you can choose that format, sorry for not pointing this out earlier, approving the pr though

**Review by subhankarshah — COMMENTED** [review]
LGTM, left a minor comment please make the change and request review from Eugene and assign to him.

**Review by subhankarshah — APPROVED** [review]
LGTM, left a minor comment please make the change and request review from Eugene and assign to him.

File: stablehlo/dialect/StablehloOps.td (hunk 3 above)
Comment (ghpvnist) [reply stablehlo/dialect/StablehloOps.td:495]: There is no pretty print available for this op currently. Thanks for the heads up though.

**Review by burmako — APPROVED** [review]

**Pull request merged**